### PR TITLE
#371: Rename COBOT_ROOT to BOT_ROOT 

### LINKED ISSUE #371: Rename COBOT_ROOT to BOT_ROOT
The env vars should be consistent with the naming system used by err, and the name "cobot" needs to die.

Need backwards compatibility for `COBOT_ROOT`, with a warning, as it will be removed after people have had a chance to change their scripts.

### PR DESCRIPTION
**Pull request by EmilyOng**
# Reviewers Checklist

- [ ] Appropriate logging is done.
- [ ] Appropriate error responses.
- [ ] Handle every possible exception.
- [ ] Make sure there is a docstring in the command functions. Hint: Lookout for
  `botcmd` and `re_botcmd` decorators.
- [ ] See that 100% coverage is there.
- [ ] See to it that mocking is not done where it is not necessary.

Closes #371 
Renamed COBOT_ROOT to BOT_ROOT

### CODE REVIEW

**Comment by mamat-rahmat** [comment]
- As stated in https://github.com/coala/corobo/issues/371#issue-276513738
  > Need backwards compatibility for COBOT_ROOT, with a warning, as it will be removed after people have had a chance to change their scripts.

- No need to create separate commit for each file as those can viewed as one change. Squash it into one commit. You can use topic instead of filename. For example : `setting`

- Commit 'subject' needs to written in imperative present tense. Change `renamed` into `rename`.

Refer to coala.io/commit

**Comment by EmilyOng** [comment]
Could you tell me a little more about the first point? Thanks 😄

File: config.py [hunk 1]
```diff
@@ -50,3 +50,11 @@
 CHATROOM_PRESENCE = os.environ.get('ROOMS', '').split() or ROOMS_TO_JOIN
 
 AUTOINSTALL_DEPS = True
+
+env_mapping = {'new-env': 'old-env'}
```
Comment (meetmangukiya) [inline config.py]: `new-env` and `old-env` were placeholders in that example. So, you should include all the environment variables that are being renamed in this form. Example: `'BOT_ROOT': 'COBOT_ROOT'`

File: config.py [hunk 2]
```diff
@@ -50,3 +50,11 @@
...
+
+for new, old in env_mapping.items():
+  if not os.environ.get(new):
+        os.environ.setdefault(new, os.environ.get(old))
+        logging.warning(
+            'Deprecation warning: environment variable {} is replaced by {}'.format(old, new))
```
Comment (meetmangukiya) [inline config.py]: Move line 53 - 60 after line 3

**Comment by yukiisbored** [comment]
@emilyong You don't have to close your PR when you're doing changes. Just mark is as wip via corobo. `corobo mark wip <pr link>`

**Comment by jayvdb** [comment]
unack https://github.com/coala/corobo/pull/378/commits/8079071bfb9618c00c7b1f64be62f485285327c8

**Comment by jayvdb** [comment]
@EmilyOng, are you going to fix this, or shall we re-assign it to someone else?

**Comment by EmilyOng** [comment]
Please do unassign, thanks

**Comment by EverWinter23** [comment]
I would like to take up this issue.

**Pull request closed without merging**